### PR TITLE
Life cycle print method

### PR DESCRIPTION
**Pull request by mylesstokowski**
This implements a pretty printing `print.life_cycle()` method and some helper functions. Note that I added a simple `format.transition` method for pretty printing an abbreviated representation of a transition - I think this makes sense within a `life_cycle`, but we might want a more verbose option for printing a transition on it's own.

### CODE REVIEW

File: R/life_cycle.R [hunk 1]
```diff
@@ -140,6 +140,37 @@ coerce_transition <- function(index, transitions) {
...
+  cat(
+    "** A life cycle",
+    "\n** Number of transitions: ", length(x),
+    "\n** Unique life stages: ", paste(life_stages(x), collapse = ", "), "\n",
+    transitions,
+    sep = ""
+  )
+}
+
+# TODO move this to transition.R and combine with any print method Dave is
```
Comment (mylesstokowski) [inline R/life_cycle.R:161]: @dallenmidd fyi

**Comment by mylesstokowski** [comment]
@dallenmidd feel free to take a look at this, but I'd like to figure out why all the automated checks are failing before merging it.

File: R/life_cycle.R (hunk 1 above)
Comment (dallenmidd) [reply R/life_cycle.R:161]: Sounds like a plan!

**Review by dallenmidd — APPROVED** [review]
Looks great to me! I will open up a new branch for making a print for transitions.

**Pull request merged**